### PR TITLE
Fixed issue with Get-AzDataProtectionBackupVault cmdlet

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Fixed issue with Get-AzDataProtectionBackupVault cmdlet

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/DataProtection/custom/Cmdlets/Platform/Vault/Get-AzDataProtectionBackupVault_list.ps1 [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿function Get-AzDataProtectionBackupVault_list
```
Comment (VeryEarly) [inline src/DataProtection/custom/Cmdlets/Platform/Vault/Get-AzDataProtectionBackupVault_list.ps1:1]: please add license, you can copy it from internal folder

File: src/DataProtection/custom/Cmdlets/Platform/Vault/Get-AzDataProtectionBackupVault_list.ps1 [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+{
+	[OutputType('PSObject')]
```
Comment (VeryEarly) [inline src/DataProtection/custom/Cmdlets/Platform/Vault/Get-AzDataProtectionBackupVault_list.ps1:3]: please double check the output type, should not be PSObject

File: src/DataProtection/README.md [hunk 3]
```diff
@@ -170,8 +170,8 @@ directive:
   - where:
       verb: Get
       subject: BackupVault
-      variant: ^GetViaIdentity2$|^Get$|^GetViaIdentity1$
-    remove: true
+      variant: ^GetViaIdentity2$|^Get$|^GetViaIdentity1$|^Get1$
```
Comment (VeryEarly) [inline src/DataProtection/README.md:173]: please hide parameter `VaultName` and `InputObject` for `Get-AzDataProtectionBackupVault` and delete the customized cmdlet

Comment (hiaga) [reply src/DataProtection/README.md:173]: This custom command was already there and was removed in PR - https://github.com/Azure/azure-powershell/pull/15774 which caused the regression. We're fixing in this PR

**Pull request merged**